microprofile/WFLY-13640_MicroProfile_Reactive_Messaging.adoc

```diff
@@ -0,0 +1,166 @@
+we have the WildFLy build consume maven artifacts from the feature pack.
+
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+It must be possible to:
+
+* deploy a Reactive Messaging application into WildFly
+* Use RSO to create/deal with streams
+* to use Context Propagation for more advanced functionality. For Context Propagation, propagation of the following will be supported:
+** CDI
+** Transactions
+** Application (currently this means the ThreadContext ClassLoader)
+
+For Reactive Messaging only constructs which are part of the RSO 1.0 specification may be used. (If we decide on allowing experimental features, we hope to introduce some features from the upcoming spec version, and possibly some SmallRye extensions. The differences can be seen in the https://github.com/eclipse/microprofile-reactive-messaging/tree/1.0/api[spec API] and https://github.com/smallrye/smallrye-reactive-messaging/tree/master/api[SmallRye API] repositories.)
+
+SmallRye Reactive Messaging comes with a set of connectors to interact with external messaging systems. We will support the following connectors:
+
+* In memory - this is not actually a connector, it is part of the core SmallRye Reactive Messaging provider
+* Kafka - used to interact with Kafka/AMQ Streams
+
+For reactive return types for REST endpoint methods, we support the following return types:
+
+* `CompletionStage` - this is part of the JDK
+* `Publisher` - This comes from the Reactive Streams specification. As mentioned in the introduction, we only support the RSO implementation of this interface.
+
+=== Organisation
+There is an extension/subsystem implementing each of the three specifications mentioned:
+
+* Reactive Messaging:
+** Extension Module: org.wildfly.extension.microprofile.reactive-messaging-smallrye
+** Subsystem Name: microprofile-reactive-messaging-smallrye
+* Context Propagation:
+** Extension Module: org.wildfly.extension.microprofile.context-propagation-smallrye
+** Subsystem Name: microprofile-context-propagation-smallrye
+* RSO:
+** Extension Module: org.wildfly.extension.microprofile.reactive-streams-operators-smallrye
+** Subsystem Name: microprofile-reactive-streams-operators-smallrye
+
+All subsystems are currently empty. In other words they have no configuration. The presence of the subsystem turns on their respective functionality.
+
+=== Galleon Layers
+The functionality is split up into layers bringing in modules and subsystems.
+
+The current list of layes can be found https://github.com/wildfly-extras/wildfly-mp-reactive-feature-pack#layers-in-this-feature-pack[here]. Note that the following layers will be removed in the product:
+
+* reactive-messaging-amqp
+* reactive-messaging-mqtt
+
+The 'convenience' layer `reactive-messaging-connectors` (in turn depended on by `microprofile-reactive-all`) will be adjusted not to include these two layers.
```

Comment:
I recommend a lot of caution about convenience layers. They are problematic because they are inherently risky for evolution, e.g. the next version adds new mega-heavy connector 'foo' and users are unhappy to suddenly have foo.  Even though you warned them in docs or whatever.

In particular because cautious about having things depend on convenience layers as that hides the evolution risk.

Reply:
I will remove these as they are not part of the backport